sys/tools/syscalls/tools/util.lua

```diff
@@ -0,0 +1,165 @@
+--
+-- SPDX-License-Identifier: BSD-2-Clause
+--
+-- Copyright (c) 2024 Tyler Baxter <agge@FreeBSD.org>
+-- Copyright (c) 2023 Warner Losh <imp@bsdimp.com>
+-- Copyright (c) 2019 Kyle Evans <kevans@FreeBSD.org>
+--
+
+local util = {}
+
+-- No char provided, trims whitespace. Char provided, trims char.
+function util.trim(s, char)
+	if s == nil then
+		return nil
+	end
+	if char == nil then
+		char = "%s"
+	end
+	return s:gsub("^" .. char .. "+", ""):gsub(char .. "+$", "")
+end
+
+-- Returns a table (list) of strings.
+function util.split(s, re)
+	local t = { }
+
+	for v in s:gmatch(re) do
+		table.insert(t, v)
+	end
+	return t
+end
+
+-- Aborts with a message and does a clean exit procedure.
+function util.abort(status, msg)
+	assert(io.stderr:write(msg .. "\n"))
+	-- cleanup
+	os.exit(status)
+end
+
+--
+-- Returns a set.
+--
+-- PARAM: t, a list
+--
+-- EXAMPLE: param: {"foo", "bar"}, return: {foo = true, bar = true}
+--
+function util.set(t)
+	local s = { }
+	for _,v in pairs(t) do s[v] = true end
+	return s
+end
+
+--
+-- Returns a set.
+--
+-- PARAM: str, a string
+-- PARAM: re, the pattern to construct keys from
+--
+function util.setFromString(str, re)
+	local s = { }
+
+	for v in str:gmatch(re) do
+		s[v] = true
+	end
+	return s
+end
+
+function util.isEmpty(tbl)
+	if tbl ~= nil then
+		if next(tbl) == nil then
+			return true
+		end
+		return false
+	end
+	return true
+end
+
+--
+--  Iterator that traverses a table following the order of its keys.
+--  An optional parameter f allows the specification of an alternative order.
+--
+--  CREDIT: https://www.lua.org/pil/19.3.html
+--  LICENSE: MIT
+--
+function util.pairsByKeys(t, f)
+	local a = {}
+	for n in pairs(t) do table.insert(a, n) end
+	table.sort(a, f)
+	local i = 0	  -- iterator variable
+	local iter = function ()   -- iterator function
+		i = i + 1
+		if a[i] == nil then return nil
+		else return a[i], t[a[i]]
+		end
+	end
+	return iter
+end
+
+--
+-- Checks for pointer types: '*', caddr_t, intptr_t.
+--
+-- PARAM: type, the type to check
+--
+-- PARAM: abi, nil or optional ABI-specified intptr_t.
+--
+function util.isPtrType(type, abi)
+	local default = "intptr_t" or abi
```

Comment:
```suggestion
	local default = abi or "intptr_t"
```
Currently default is always set to `"intptr_t"`